### PR TITLE
Various minor improvements

### PR DESCRIPTION
**Pull request by tortmayr**
- Fix for failing Travis deployment on master branch
- Ensure proper cleanup of test workspace
- Swap logger library to lf4j-log4j12 to ensure consistent logging for all components
- Enabled fatjar generation per Default. (This could be an issue for p2 generation but the profile can be deactivated if needed)

- Use commons CLI library for cli argument parsing
- Add options for setting workspace root and loglevel via command line arguments

NOTE: We need to improve our Logging configuration at some point. Currently, the Logger just forwards to System.Out. Ideally, errors should be printed to System.error so I can react in the Theia Backend by listening to the error stream. However, for now, I bypassed this problem by setting the log level to error via CLI argument.

### CODE REVIEW

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/CLIParser.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import java.util.Optional;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+public class CLIParser {
+	public static int DEFAULT_SERVER_PORT = 8081;
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/CLIParser.java]: Can we re-use `ModelServerLauncher.DEFAULT_JAVALIN_PORT`?

File: examples/com.eclipsesource.modelserver.example/src/main/java/com/eclipsesource/modelserver/example/ExampleServerLauncher.java [hunk 2]
```diff
@@ -35,23 +39,37 @@
 
 	private static Logger LOG = Logger.getLogger(ExampleServerLauncher.class);
 
-	public static void main(String[] args) {
+	public static void main(String[] args) throws ParseException {
 		BasicConfigurator.configure();
+		CLIParser.create(args, CLIParser.getDefaultCLIOptions());
+		Logger root = Logger.getRootLogger();
+		if (CLIParser.getInstance().optionExists("e")) {
```
Comment (edgarmueller) [inline examples/com.eclipsesource.modelserver.example/src/main/java/com/eclipsesource/modelserver/example/ExampleServerLauncher.java:46]: Reading the CLI options seems as if it should be part of `ModelServerLauncher` (as we do with the port). Is there a reason why this is located here?

**Review by edgarmueller — COMMENTED** [review]
Looks good! I have left a question and minor remark.

File: examples/com.eclipsesource.modelserver.example/src/main/java/com/eclipsesource/modelserver/example/ExampleServerLauncher.java (hunk 2 above)
Comment (tortmayr) [reply examples/com.eclipsesource.modelserver.example/src/main/java/com/eclipsesource/modelserver/example/ExampleServerLauncher.java:46]: Technically you are right (at least the CLI option for logging can be moved into the ModelServerLauncher).

We still need to read the CLI options here to check the workspace root. If no root was specified we use the testworkspace as a fallback (and modify the launch args accordingly)

Comment (tortmayr) [reply examples/com.eclipsesource.modelserver.example/src/main/java/com/eclipsesource/modelserver/example/ExampleServerLauncher.java:46]: Moving the logLevel CI-check to the `ModelServerLauncher` doesn't work properly. Some debug logs slip through.

Maybe we should leave this as is for the moment. I will create an issue for improving the log configuration (i.e Errors are tracked exclusively to the error stream). Once this issue is resolved the cli loglevel configuration could be removed entirely.

**Review by edgarmueller — APPROVED** [review]
Thank you!

**Pull request merged**